### PR TITLE
Refactor velocity postprocessor output

### PR DESCRIPTION
**Pull request by gassmoeller**
This refactors some of our velocity statistics postprocessors to remove code duplication caused by unit conversion between seconds and years. Also assert a special behavior of the boundary_velocity_residual_statistics postprocessor, which looks like it is prepared to work for several boundaries simultaneously, which we did not test so far (currently it is hard-coded to only compute the statistics for the surface boundary).

### CODE REVIEW

File: source/postprocess/boundary_velocity_residual_statistics.cc [hunk 1]
```diff
@@ -221,89 +225,49 @@ namespace aspect
...
+      const double unit_scale_factor = (this->convert_output_to_years() == true) ? year_in_seconds : 1.0;
       std::ostringstream screen_text;
       unsigned int index = 0;
+
       for (std::map<types::boundary_id, double>::const_iterator
            p = global_max_vel.begin(), a = global_min_vel.begin(), rms = global_rms_vel.begin();
            p != global_max_vel.end() && a != global_min_vel.end() && rms != global_rms_vel.end();
            ++p, ++a, ++rms, ++index)
         {
-          if (this->convert_output_to_years() == true)
```
Comment (tjhei) [inline source/postprocess/boundary_velocity_residual_statistics.cc:231]: Do you know what the meaning of p and a is? Can you replace the variable names please?

**Comment by tjhei** [comment]
> 457 tests failed out of 914
This is a candidate for this year's bucket award. 😉

**Review by tjhei — APPROVED** [review]
Looks good otherwise!

**Comment by gassmoeller** [comment]
>  This is a candidate for this year's bucket award. wink

Yes, that is the risk when fiddling around with one of the oldest postprocessors :-). Let's see what the tester thinks about this new version.

**Review by tjhei — APPROVED** [review]

**Pull request merged**